### PR TITLE
Formatter: consider consecutive macro literals when subformatting

### PR DESCRIPTION
**Pull request by asterite**
The lexer might produce consecutive macro literals instead of just a single one, so when subformatting consider this case too.

Spotted by @waj

### CODE REVIEW

File: src/llvm/enums.cr [hunk 1]
```diff
@@ -2,70 +2,70 @@ module LLVM
...
+      SwiftSelf                   = 1 << 48
+      UWTable                     = 1 << 49
+      WriteOnly                   = 1 << 50
+      ZExt                        = 1 << 51
 
       @@kind_ids = load_llvm_kinds_from_names.as(Hash(Attribute, UInt32))
 
       def each_kind(&block)
         return if value == 0
-        \{% for member in @type.constants %}
```
Comment (Sija) [inline src/llvm/enums.cr]: That removal seems like a bug.

Comment (asterite) [reply src/llvm/enums.cr]: Thanks, hopefully fixed! 🤞

File: src/compiler/crystal/tools/formatter.cr [hunk 2]
```diff
@@ -2020,6 +2031,17 @@ module Crystal
       false
     end
 
+    # If we are formatting macro contents, if there are nested macro
+    # control structures they are definitely escaped with `\`,
+    # because otherwise we wouln't be able to format the contents.
+    # So here we appent those slashes. In theory the nesting can be
```
Comment (Sija) [inline src/compiler/crystal/tools/formatter.cr]: typo: append

File: src/compiler/crystal/tools/formatter.cr [hunk 3]
```diff
@@ -2020,6 +2031,17 @@ module Crystal
...
+    # very deep but it's usually just one nesting.
```
Comment (Sija) [inline src/compiler/crystal/tools/formatter.cr]: nesting -> level

File: src/compiler/crystal/tools/formatter.cr (hunk 2 above)
Comment (asterite) [reply src/compiler/crystal/tools/formatter.cr]: Thanks, fixed

File: src/compiler/crystal/tools/formatter.cr (hunk 3 above)
Comment (asterite) [reply src/compiler/crystal/tools/formatter.cr]: Thanks, fixed

**Pull request merged**